src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -537,13 +600,30 @@ private IAIState helping()
     }
 
     /**
-     * Decide what we should do next! Ticked once every 20 Ticks
+     * Decide what we should do next! Ticked once every GUARD_TASK_INTERVAL Ticks
      *
      * @return the next IAIState.
      */
     protected IAIState decide()
     {
         reduceAttackDelay(GUARD_TASK_INTERVAL * getTickRate());
+
+        final ILocation rallyLocation = buildingGuards.getRallyLocation();
+
+        if (rallyLocation != null || buildingGuards.getTask() == GuardTask.FOLLOW)
+        {
+            worker.addPotionEffect(new EffectInstance(GLOW_EFFECT, GLOW_EFFECT_DURATION, GLOW_EFFECT_MULTIPLIER, false, false));
```

Comment:
think we should check if a potion effect already exists before applying a new one

Reply:
The Follow-code didn't do that either, and as far as I understand, adding potion effects ontop of existing ones will simply extend the duration.
So, as far as I understand it, checking for the potion effect will only cost performance without doing anything meaningful.

Reply:
actually the applying is more expensive than checking for existing, since applying involves events fired for other mods which listen to potion stuff. And some of those mods checks are rather laggy, thats why xD

Reply:
I understand, but simply adding the if-clause would cause stutter-glowing. The decide() function is only called once every 5 seconds. If the effect runs out 1 second after the last check, the guard will stop glowing for 4 seconds, and then light up again. Is it worth adding logic to check the remaining time of the effect vs. the time until the next check?

Reply:
once every 5 seconds should be finne.